llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -423,24 +423,17 @@ void llvm::computeKnownBitsFromRangeMetadata(const MDNode &Ranges,
   unsigned NumRanges = Ranges.getNumOperands() / 2;
   assert(NumRanges >= 1);
 
-  Known.Zero.setAllBits();
-  Known.One.setAllBits();
+  ConstantRange CR = ConstantRange::getFull(BitWidth);
```

Comment:
```suggestion
  ConstantRange CR = ConstantRange::getEmpty(BitWidth);
```

It should be initialized to empty :)